tests/test-ucx-w-dask-workers.py

```diff
@@ -0,0 +1,60 @@
+import pytest
+from distributed.utils import format_bytes
+import numpy as np
+import asyncio
+import dask.dataframe as dd
+from distributed import Scheduler, Worker, Client, Nanny
+from distributed.utils import log_errors
+
+#async with Nanny(s.address, protocol='ucx', nthreads=1,
+# nanny is really a worker running on a defined CUDA DEVICE
+protocol = 'ucx'
+interface = 'ib3'
+async def f(cudf_obj):
+    async with Scheduler(protocol=protocol, interface=interface,
+    dashboard_address=':8789') as s:
+        async with Nanny(s.address, protocol=protocol, nthreads=1,
+                memory_limit='32GB', interface=interface,
+                env={'CUDA_VISIBLE_DEVICES': '6,7'},
+                ) as w:
+            async with Nanny(s.address, protocol=protocol,memory_limit='32gb',
+                    env={'CUDA_VISIBLE_DEVICES': '7,6'}, interface=interface,
+                    nthreads=1) as w2:
+                async with Client(s.address, asynchronous=True) as c:
+                    with log_errors():
+
+                        def set_nb_context(x=None):
+                            import numba.cuda
+                            try:
+                                numba.cuda.current_context()
+                            except Exception:
```

Comment:
I don't see CUDA_VISIBLE_DEVICES (CVD) being set for the client explicitly. Is that intended? If yes, does the test expect client to make allocations on device 0 (which will be default if not explicitly set) and have this be communicated to workers/scheduler? Also what is CVD set to in the case of scheduler?
IMO, the client, worker(s), scheduler should all have the same consistent view of CUDA capable devices -- so either set CVD for all entities or set for none.

Reply:
The client is in the main process, so if we're using environment variables then this will have to be handled by the user calling pytest like the following:

```
CUDA_VISIBLE_DEVICES=0,1 py.test foo.py
```

Alternatively, are there ways to set this programmatically within UCX?